tests/ui-api-layer-acceptance-tests/domain/remoteenvironment/remoteenvironment_test.go

```diff
@@ -22,64 +36,153 @@ type reDeleteMutation struct {
 }
 
+type remoteEnvironmentEvent struct {
+	Type              string
+	RemoteEnvironment remoteEnvironment
+}
+
 func TestRemoteEnvironmentMutations(t *testing.T) {
 	if dex.IsSCIEnabled() {
 		t.Skip("SCI Enabled")
 	}
 	c, err := graphql.New()
 	require.NoError(t, err)
 
+	reCli, _, err := client.NewREClientWithConfig()
+	require.NoError(t, err)
+
+	t.Log("Subscribe On Remote Environments")
+	subscription := subscribeREEvent(c)
+	defer subscription.Close()
+
 	const fixName = "test-ui-api-re"
-	fixResponse := reMutationData{
-		Name:        fixName,
-		Description: "fix-desc1",
-		Labels: map[string]string{
-			"fix": "lab",
-		},
-	}
-	t.Log("Create RemoteEnvironment")
-	resp, err := createRE(c, fixResponse)
+	fixedRE := createREStruct(fixName, "fix-desc1", map[string]string{"fix": "lab"})
+
+	waitForREReady(fixName, reCli)
+
+	t.Log("Create Remote Environment")
+	resp, err := createRE(c, fixedRE)
 	require.NoError(t, err)
-	assert.Equal(t, fixResponse, resp.ReCreateMutation)
+	assert.Equal(t, fixedRE, resp.ReCreateMutation)
+
+	t.Log("Check Subscription Event")
+	expectedEvent := createREEventStruct("ADD", fixedRE)
+	event, err := readREEvent(subscription)
+	assert.NoError(t, err)
+	checkREEvent(t, expectedEvent, event)
 
 	defer func() {
-		t.Log("Delete RemoteEnvironment")
+		t.Log("Delete Remote Environment")
 		deleteResp, err := deleteRE(c, fixName)
 		require.NoError(t, err)
 		assert.Equal(t, fixName, deleteResp.ReDeleteMutation.Name)
 	}()
 
-	t.Log("Update RemoteEnvironment")
-	fixResponse.Description = "desc2"
-	fixResponse.Labels = map[string]string{
-		"lab": "fix",
-	}
-	updateResp, err := updateRE(c, fixResponse)
+	t.Log("Update Remote Environment")
+	fixedRE = createREStruct(fixName, "desc2", map[string]string{"lab": "fix"})
+
+	updateResp, err := updateRE(c, fixedRE)
 	require.NoError(t, err)
-	assert.Equal(t, fixResponse, updateResp.ReUpdateMutation)
+	assert.Equal(t, fixedRE, updateResp.ReUpdateMutation)
 }
 
-func createRE(c *graphql.Client, given reMutationData) (reCreateMutationResponse, error) {
-	query := `
+func createREStruct(name string, desc string, labels map[string]string) remoteEnvironment {
+	return remoteEnvironment{
+		Name:        name,
+		Description: desc,
+		Labels:      labels,
+	}
+}
+
+func readREEvent(sub *graphql.Subscription) (remoteEnvironmentEvent, error) {
+	type Response struct {
+		RemoteEnvironmentEvent remoteEnvironmentEvent
+	}
+	var reEvent Response
+	err := sub.Next(&reEvent, tester.DefaultSubscriptionTimeout)
+
+	return reEvent.RemoteEnvironmentEvent, err
+}
+
+func subscribeREEvent(c *graphql.Client) *graphql.Subscription {
+	query := fmt.Sprintf(`
+			subscription {
+				%s
+			}
+		`, reEventFields())
+	req := graphql.NewRequest(query)
+	return c.Subscribe(req)
+}
+
+func reEventFields() string {
+	return fmt.Sprintf(`
+        remoteEnvironmentEvent {
+			type
+    		remoteEnvironment{
+				%s
+			}
+        }
+    `, reFields())
+}
+
+func reFields() string {
+	return `
+		name
```

Comment:
As I wrote before: make sure that you return all fields of RE type. They are defined here: https://github.com/kyma-project/kyma/blob/master/components/ui-api-layer/internal/gqlschema/schema.graphql